lib/streamlit/runtime/state/session_state.py

```diff
@@ -298,6 +299,9 @@ class SessionState:
     # Keys used for widgets will be eagerly converted to the matching widget id
     _key_id_mapping: dict[str, str] = field(default_factory=dict)
 
+    # query params are stored in session state because query params will be tied with widget state at one point.
+    _query_params: QueryParams = field(default_factory=QueryParams)
```

Comment:
nit: if we're expecting to access `query_params` from outside of the `SessionState` class (which seems to be the case from subsequent PRs), then it shouldn't have its name prefixed with an underscore